### PR TITLE
feat: Add Carousel touch support

### LINKED ISSUE #3683: Add touch support for carousels

### PR DESCRIPTION
**Pull request by bpas247**
Touch support should be provided for our Carousel component, as
per the [upstream docs for Carousel](https://getbootstrap.com/docs/4.3/components/carousel/#options) (as of Bootstrap v4.2).

Fixes #3683.

### CODE REVIEW

**Comment by jquense** [comment]
Doesn't look like upstream uses Hammer.js what do you think about implementing this with straight native events like they do instead of adding a new dependency?

File: test/CarouselSpec.js [hunk 1]
```diff
@@ -287,4 +287,24 @@ describe('<Carousel>', () => {
...
+
+    try {
+      const onSelectSpy = sinon.spy();
+      const wrapper = mount(
+        <Carousel interval={0} onSelect={onSelectSpy} touch>
+          {items}
+        </Carousel>,
+      );
+
+      wrapper.simulate('touchStart', { changedTouches: [{ screenX: 0 }] });
```
Comment (bpas247) [inline test/CarouselSpec.js]: Hmm, can't seem to figure out why this isn't firing the touch event handler...

Comment (jquense) [reply test/CarouselSpec.js]: Are you manually attaching touch handlers or using the react event system? Enzyme will simulate at the react level, but not actually fire Dom events

File: src/Carousel.js [hunk 2]
```diff
@@ -221,6 +230,39 @@ class Carousel extends React.Component {
...
+    // If the swipe is under the threshold, don't do anything.
+    if (
+      Math.abs(e.changedTouches[0].screenX - this.state.touchStartX) <
+      SWIPE_THRESHOLD
+    )
+      return;
+
+    // Going left
+    if (e.changedTouches[0].screenX < this.state.touchStartX) {
+      this.to(
```
Comment (taion) [inline src/Carousel.js]: does it make sense to use `handlePrev` and `handleNext` here?

**Comment by xyzr07** [comment]
thank you

**Pull request merged**